daal4py/sklearn/test/test_run_to_run_stability_tests.py

```diff
@@ -36,300 +33,231 @@
 from sklearn.cluster import KMeans
 from sklearn.cluster import DBSCAN
 from sklearn.decomposition import PCA
-
 from sklearn.svm import SVC
 from sklearn.datasets import make_classification
 from sklearn.datasets import make_regression
+from sklearn.base import is_classifier, is_regressor
 
-##################################################
 
-def run_clf_test(func):
-    for features in [5, 10]:
-        X, y = make_classification(n_samples=4000, n_features=features, n_informative=features, n_redundant=0,
-            n_clusters_per_class=8, random_state=0)
+# to reproduce errors even in CI
+d4p.daalinit(nthreads=100)
 
-        baseline, name = func(X, y)
 
-        for i in range(10):
-            res, _ = func(X, y)
+def get_class_name(x):
+    return x.__class__.__name__
+
+
+WITHOUT_FIT = [
+    'DBSCAN',
+]
+
+
+def func(X, Y, model, attribute, methods):
+    clf = model
+    if get_class_name(model) not in WITHOUT_FIT:
+        clf.fit(X, Y)
+
+    res = []
+    name = []
+
+    for i in methods:
+        if i == 'predict':
+            res.append(clf.predict(X))
+            name.append(get_class_name(model) + '.predict(X)')
+        elif i == 'predict_proba':
+            res.append(clf.predict_proba(X))
+            name.append(get_class_name(model) + '.predict_proba(X)')
+        elif i == 'kneighbors':
+            dist, idx = clf.kneighbors(X)
+            res.append(dist)
+            name.append('dist')
+            res.append(idx)
+            name.append('idx')
+        elif i == 'fit_predict':
+            predict = clf.fit_predict(X)
+            res.append(predict)
+            name.append(get_class_name(model) + '.fit_predict')
+        elif i == 'transform':
+            res.append(clf.transform(X))
+            name.append(get_class_name(model) + '.transform(X)')
+        elif i == 'get_covariance':
+            res.append(clf.get_covariance())
+            name.append(get_class_name(model) + '.get_covariance()')
+        elif i == 'get_precision':
+            res.append(clf.get_precision())
+            name.append(get_class_name(model) + '.get_precision()')
+        elif i == 'score_samples':
+            res.append(clf.score_samples(X))
+            name.append(get_class_name(model) + '.score_samples(X)')
+
+    for i in attribute:
+        res.append(getattr(clf, i))
+        name.append(get_class_name(model) + i)
+    return res, name
 
-            for a, b, n in zip(res, baseline, name):
-                np.testing.assert_allclose(a, b, rtol=0.0, atol=0.0, err_msg=str(n + " is incorrect"))
 
-def run_reg_test(func):
+def _run_test(model, attribute, methods):
     for features in [5, 10]:
-        X, y = make_regression(n_samples=4000, n_features=features, n_informative=features, random_state=0, noise=0.2, bias=10)
-        baseline, name = func(X, y)
+        if is_classifier(model) or get_class_name(model) == 'DBSCAN':

```

Comment:
DBSCAN and KMeans are cluster algorithms. Why is there only dbscan? can use for them `make_blobs`?